### DISCUSSION TITLE
THANK YOU! Recursion is the thing that sold bun for me

### DISCUSSION THREAD

**Discussion by the-main-thing** [oven-sh/bun#1889] [Show and tell]
I love using recursion. But in node this is dangerous, because there is a limit for recursion depth.
In bun I just did more than 130 000 levels deep recursion and it is amazing!
I don't usually need that much deepness, but it is nice to see that I am able not to worry and just use the tool I like.

Thank you!

**Comment by the-main-thing** [comment]
Oh, nevermind)) I just ran some actual code, not synthetic test. And it failed with max call stack.

**Comment by kamalciirs** [comment]
```js
const rec = depth => {
    depth % 1e9 === 0 && console.log(depth)
    return rec(++depth)
}

rec(0)
```

```powershell
PS C:\Users\...> bun app.js
0
1000000000
2000000000
3000000000
...
99000000000
100000000000
101000000000
102000000000
...
```

Bro is still going..
